Enable conditional `source` and `use` patterns by allowing `null` as a no-op module

Bahex opened the pull request:
Related:
- #14329
- #13872
- #8214

# Description & User-Facing Changes

This PR allows enables the following uses, which are all no-op.
```nushell
source null
source-env null
use null
overlay use null
```

The motivation for this change is conditional sourcing of files. For example, with this change `login.nu` may be deprecated and replaced with the following code in `config.nu`
```nushell
const login_module = if $nu.is-login { "login.nu" } else { null }
source $login_module
```

# Tests + Formatting
I'm hoping for CI to pass :smile:

# After Submitting
Add a part about the conditional sourcing pattern to the website.

fdncred commented on crates/nu-cmd-lang/src/core_commands/use_.rs:
```diff
@@ -22,7 +22,11 @@ impl Command for Use {
+            .required(
+                "module",
+                SyntaxShape::OneOf(vec![SyntaxShape::String, SyntaxShape::Nothing]),
+                "Module or module file.",
```
I wonder if this should be something like `Module, module file, or null for the no-op use case`? Same for overlay use, and the source commands.

Bahex replied:
I should add examples to each command before this is merged.

fdncred reviewed (approved):
Just a minor question. This looks better than mine, even though this doesn't seem to handle open. I'm fine with starting here. Nice work!

I'm assuming you can get the CI green too. :)

fdncred reviewed (commented):
Maybe I shouldn't approve it with the CI not green. LOL

Bahex commented:
Turned out to be extremely trivial to fix, I messed up `source-env`'s signature

fdncred commented:
Ah, interesting.

Looking good. Ping me when it's ready.

sholderbach commented:
Would you still be able to import string `"null"` as a module/file name or is that blocked for other reasons?

Bahex commented:
@sholderbach works without problem

```
nu ❯❯ ls **/*
╭─#─┬────name─────┬─type─┬─size─┬───modified────╮
│ 0 │ null        │ dir  │ 60 B │ 2 minutes ago │
│ 1 │ null/mod.nu │ file │ 31 B │ 2 minutes ago │
╰─#─┴────name─────┴─type─┴─size─┴───modified────╯
nu ❯❯ open null/mod.nu
export-env {
    print "Hello!"
}
nu ❯❯ use null
nu ❯❯ use "null"
Hello!
```

Bahex commented:
@fdncred I've added some tests, and added examples to `source` commands.

I didn't add examples to `use` and `overlay use` as this change is mainly useful with `source`, and their examples are already quite crowded.
Should I add examples to `use` commands too just for consistency?

fdncred commented:
@Bahex I think we could skip the example if we just add more info in the parameter docs like I said above. Such as
```
Module, module file, or null for the no-op
```
If we do that on the parameters, I think it would help.

fdncred commented:
Thanks. Nice work!

fdncred commented:
It would be great to showcase this in our upcoming release notes and documentation somewhere as well.
